app/src/main/res/navigation/mobile_navigation.xml

```diff
@@ -198,6 +201,14 @@
                 android:defaultValue="@null"
                 app:argType="string" />
         </action>
+        <action
+            android:id="@+id/action_nav_favorList_to_chatView"
+            app:destination="@id/chatView">
+            <argument
+                android:name="FAVOR_ID"
+                android:defaultValue="@null"
```

Comment:
I think a simpler solution in the end will be to pass the favor object here (that way we don't need to use the viewholder or the observer in the chat, which can give us performance problems.